api/api/serializers/vulns.py

```diff
@@ -22,3 +22,32 @@ def update(self, instance, validated_data):
         if "last_updated_date" not in validated_data:
             validated_data["last_updated_date"] = datetime.now()
         return super().update(instance, validated_data)
+
+class VulnTypeSerializer(serializers.ModelSerializer):
+    class Meta:
+        fields = [
+            'id', 'name', 'description'
+        ]
+        model = VulnType
+
+
+class VulnerabilitySerializer(serializers.ModelSerializer):
+    class Meta:
+        fields = [
+            'id', 'title', 'description', 'images', 'author', 'last_editor', 'vuln_type'
+        ]
+
+    def create(self, validated_data):
+        if "creation_date" not in validated_data:
+            validated_data["creation_date"] = datetime.now()
+            validated_data["last_updated_date"] = datetime.now()
+        if "last_editor" not in validated_data:
+            validated_data["last_editor"] = validated_data["author"]
+        validated_data["vuln_type"] = VulnType.objects.filter(name=validated_data["vuln_type"]).id
```

Comment:
why not use Nested Serializers instead ? You can find an example in `AdminSerializer`

Reply:
It's not the same because it's super super rare that you'll create a new vulnerability type, so the two forms won't be the same don't you think ?